src/components/CodeSample.vue

```diff
@@ -0,0 +1,96 @@
+<template>
+  <div class="syn-code-block">
+    <div class="syn-code-block-header syn-flex justify-space-between">
+      <span class="syn-overline syn-mb-0">{{title}}</span>
+      <div class="syn-ma-8 syn-flex align-center">
+        <select class="syn-code-language-selector syn-mr-8" v-if="languages.length > 1" v-model="selectedLanguage">
+          <option v-for="language in languages" :key="language" :value="language">
+            {{ language }}
+          </option>
+        </select>
+        <button class="syn-btn no-padding" @click="copySample" title="Copy">
+          <svg width="14" height="14" viewBox="0 0 24 24" class="syn-ma-4">
+            <path
+              fill="currentColor"
+              d="M16 1H4c-1.1 0-2 .9-2 2v14h2V3h12V1zm-1 4H8c-1.1 0-1.99.9-1.99 2L6 21c0 1.1.89 2 1.99 2H19c1.1 0 2-.9 2-2V11l-6-6zM8 21V7h6v5h5v9H8z"
+            />
+          </svg>
+        </button>
+      </div>
+    </div>
+    <div class="syn-code-block-body" ref="codeContent">
+      <slot>
+        <pre><code>// Sample Code Block
+  console.log('Hello world!')</code></pre>
+      </slot>
+    </div>
+  </div>
+</template>
+
+<script>
+export default {
+  name: 'CodeSample',
+  props: {
+    title: {
+      type: String,
+      default: ''
+    }
+  },
+  data() {
+    return {
+      languages: [],
+      selectedLanguage: ''
+    }
+  },
+  methods: {
+    copySample() {
+      const copyText = this.$refs.codeContent.querySelector(`figure.${this.selectedLanguage}`).innerText
+      const el = document.createElement('textarea')
+      el.value = copyText
+      document.body.appendChild(el)
+      el.select()
+      document.execCommand('copy')
+      document.body.removeChild(el)
```

Comment:
I imagine as we add more functionality to the docs that this snippet would be a very useful little helper function. Mind if we  create `copyToClipboard()` in a `utilities` folder?

Reply:
Sounds good!